command/metadata.go

```diff
@@ -0,0 +1,125 @@
+package command
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mackerelio/mackerel-agent/config"
+	"github.com/mackerelio/mackerel-agent/metadata"
+)
+
+func metadataGenerators(conf *config.Config) []*metadata.Generator {
+	generators := make([]*metadata.Generator, 0, len(conf.MetadataPlugins))
+
+	workdir := os.Getenv("MACKEREL_PLUGIN_WORKDIR")
+	if workdir == "" {
+		workdir = os.TempDir()
```

Comment:
`os.TempDir()` is shared with other processes, so putting many files directly under TempDir looks not good, just imo.